common/persons_api/api.go

```diff
@@ -13,6 +15,9 @@ type Client struct {
 	accessToken  string
 	httpClient   *http.Client
 	baseUrl      string
+	authUrl      string
+
+	rwLock *sync.RWMutex
```

Comment:
Just as a side note -- you can also have a struct inherit RWMutex behavior. e.g.:
```
type Client struct {
        sync.RWMutex // inherit RW Lock behavior
  	accessToken  string
	httpClient   *http.Client
	baseUrl      string
	authUrl      string
}
```

Then you can use it as:

```
c.Lock()
defer c.Unlock()
```

Reply:
Yeah, but that's a bit confusing and exposes those funcs outside of the package.